Add a note about best practice for choice of a backup store. (#780)
longhorn/website: content/docs/1.4.0/best-practices.md

## Volume Maintenance

> Using Longhorn's built-in backup feature is highly recommended. You can save backups to an object store (such as S3 and Azure), an NFS server, or an SMB or CIFS server. Saving to an object store is preferable.

Reviewer: > Saving to an object store is preferable.

It would be nice if you could comment on the advantages of using an object store.

> Using Longhorn's built-in backup feature is highly recommended. You can save backups to an object store such as S3 or to an NFS server. Saving to an object store is preferable because it generally offers better performance and reliability.  Another advantage is that you do not need to mount and unmount the target, which can complicate failover and upgrades.

Reviewer: The statement of "better performance" is probably incorrect. I would suggest removing it and only keeping "better reliability".

> Using Longhorn's built-in backup feature is highly recommended. You can save backups to an object store such as S3, or an NFS server. Saving to an object store is preferable, if possible, because it generally has better performance and reliability.  There is also the advantage of not having to mount and unmount the target, which can complicate failover and upgrades.

Reviewer 2: ```suggestion
Using Longhorn's built-in backup feature is highly recommended. You can save backups to an object store such as S3 or to an NFS server. Saving to an object store is preferable because it generally offers better performance and reliability. Another advantage is that you do not need to mount and unmount the target, which can complicate failover and upgrades.
```

Author: Done.
